### PR TITLE
rgw/multisite: don't retain RGW_ATTR_OBJ_REPLICATION_TRACE attr on copy_object

### PR DESCRIPTION
**Pull request by smanjara**
if a copy_object is executed on a replicated object, it fails to sync the copied object to the original source zone where the source object was first written and replicated from. this is because copy_object retains source attrs by default. when a get_obj() is issued from a fetch_remote_obj() call during sync, we check for this RGW_ATTR_OBJ_REPLICATION_TRACE and if that destination zone is already present in the trace, we return NOT_MODIFIED error, thus failing to replicate the copied object.

https://tracker.ceph.com/issues/66983

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_rados.cc [hunk 1]
```diff
@@ -4724,6 +4724,8 @@ int RGWRados::copy_obj(RGWObjectCtx& obj_ctx,
     src_attrs.erase(RGW_ATTR_OLH_VER);
   }
 
+  src_attrs.erase(RGW_ATTR_OBJ_REPLICATION_TRACE);
```
Comment (cbodley) [inline src/rgw/driver/rados/rgw_rados.cc:4727]: should we clear RGW_ATTR_OBJ_REPLICATION_STATUS and RGW_ATTR_OBJ_REPLICATION_TIMESTAMP too?

**Review by cbodley — COMMENTED** [review]
if we don't have any test coverage for copy_object(), could you add a simple one?

File: src/rgw/driver/rados/rgw_rados.cc (hunk 1 above)
Comment (smanjara) [reply src/rgw/driver/rados/rgw_rados.cc:4727]: yes, done.

**Comment by smanjara** [comment]
jenkins retest

File: src/rgw/driver/rados/rgw_rados.cc (hunk 1 above)
Comment (cbodley) [reply src/rgw/driver/rados/rgw_rados.cc:4727]: RGW_ATTR_OBJ_REPLICATION_STATUS can go too

**Review by cbodley — APPROVED** [review]

**Comment by ivancich** [comment]
Teuthology results, no regressions:
* https://pulpito.ceph.com/ivancich-2024-07-15_15:27:15-rgw-wip-eric-testing-2-distro-default-smithi/
* https://pulpito.ceph.com/ivancich-2024-07-15_15:26:59-rgw-wip-eric-testing-2-distro-default-smithi/

**Comment by ivancich** [comment]
jenkins test make check

**Comment by ivancich** [comment]
jenkins test windows

**Comment by ivancich** [comment]
@smanjara This passed QA and when the Jenkins jobs end I'd be ready to merge.

Will this need backports? If so, please add an appropriate tracker.

**Comment by smanjara** [comment]
> @smanjara This passed QA and when the Jenkins jobs end I'd be ready to merge.
> 
> Will this need backports? If so, please add an appropriate tracker.

@ivancich thanks Eric. yes, it needs to be backported. created tracker [66983](https://tracker.ceph.com/issues/66983)

**Comment by ivancich** [comment]
jenkins test make check

**Comment by ivancich** [comment]
jenkins test make check

**Pull request merged**